python/lsst/analysis/ap/apdb.py

```diff
@@ -66,13 +78,65 @@ def connection(self):
         """
         pass
 
-    def load_sources_for_object(self, dia_object_id, limit=100000):
+    def set_bad_DiaSource_flags(self, flag_list):
+        """Set DiaSource flags to exclude.
+
+        Parameters
+        ----------
+        flag_list : `list` [`str`]
+            Flag names to exclude
+        """
+
+        flag_map = os.path.join(lsst.utils.getPackageDir("ap_association"),
+                                "data/association-flag-map.yaml")
+        unpacker = UnpackApdbFlags(flag_map, "DiaSource")
+
+        for flag in flag_list:
+            if flag not in [c[0] for c in unpacker.output_flag_columns['flags']]:
```

Comment:
This feels a bit like poking at the innards of the unpacker. We should probably add an `unpacker.is_valid(flag, columnName)` method to the unpacker (not necessarily on this ticket?).

Reply:
I went ahead and created `unpacker.flagExists` because it helped with some of the checks in `ap_association` (and revealed bugs!).